tests/unit/models/ObjectMD.js

```diff
@@ -257,6 +257,25 @@ describe('ObjectMD class setters/getters', () => {
         md.clearMetadataValues();
         assert.strictEqual(md.getUserMetadata(), undefined);
     });
+
+    it('ObjectMD::microVersionId unset', () => {
+        assert.strictEqual(md.getMicroVersionId(), null);
+    });
+
+    it('ObjectMD::microVersionId set', () => {
+        const generatedIds = new Set();
+        for (let i = 0; i < 100; ++i) {
+            md.updateMicroVersionId();
+            generatedIds.add(md.getMicroVersionId());
+        }
+        // all generated IDs should be different
+        assert.strictEqual(generatedIds.size, 100);
+        generatedIds.forEach(key => {
```

Comment:
Could checking length of `microVersionId` be a separate test? It will be cleaner!

Reply:
IMHO this is a good place to check that the length of all entries is good as well as we just generated the set of version IDs. I don't think it deserves a test on its own.